examples/nas/oneshot/spos/search.py

```diff
@@ -1,17 +1,88 @@
-# This file is to demo the usage of multi-trial NAS in the usage of SPOS search space.
-
-import click
+import time
 import json
-import nni.retiarii.evaluator.pytorch as pl
+import random
+import logging
+import argparse
+import numpy as np
+import torch
+import torchvision.transforms as transforms
+import torchvision.datasets as datasets
+
+import nni
+from nn_meter import load_latency_predictor
+import nni.retiarii.nn.pytorch as nn
 import nni.retiarii.strategy as strategy
-from nni.retiarii import serialize
+from nni.retiarii.evaluator.functional import FunctionalEvaluator
+from nni.retiarii.utils import original_state_dict_hooks
+from nni.retiarii.oneshot.pytorch.utils import AverageMeterGroup
 from nni.retiarii.experiment.pytorch import RetiariiExeConfig, RetiariiExperiment
-from torchvision import transforms
-from torchvision.datasets import CIFAR10
-from nn_meter import load_latency_predictor
 
-from network import ShuffleNetV2OneShot
-from utils import get_archchoice_by_model
+from network import ShuffleNetV2OneShot, load_and_parse_state_dict
+from utils import CrossEntropyLabelSmooth, accuracy, ToBGRTensor, get_archchoice_by_model
```

Comment:
Why do we need this?

get_archchoice_by_model

Reply:
I found the top model exported by `exp.export_top_models()` is a `dict` with format as "LayerChoice1": "layerchoice_LayerChoice1_0". However when loading architecture from `fixed_arch` the dict value is needed to be converted to int. Such that I removed 'layerchoice' in choice value to meet the requirement of `fixed_arch`. The output of `get_archchoice_by_model` will be "LayerChoice1": "0".

Reply:
Did you try python execution engine? Does it work properly?

Reply:
I tried python execution. The exported model of python engine is "LayerChoice1": "0" already. So that `get_archchoice_by_model` will do nothing to the string "0" in Dict and the output of `get_archchoice_by_model` is the same as original inputs. The script won't raise any error.

Reply:
Suggest adding some comments on `get_archchoice_by_model`. You can simply copy your response in the PR to the code comments.

Reply:
Have fixed it.